pkg/protocol/protocol_commitments.go

```diff
@@ -0,0 +1,104 @@
+)
+
+// CommitmentsProtocol is a subcomponent of the protocol that is responsible for handling commitment requests and
+// responses.
+type CommitmentsProtocol struct {
+	// protocol contains a reference to the Protocol instance that this component belongs to.
+	protocol *Protocol
+
+	// workerPool contains the worker pool that is used to process commitment requests and responses asynchronously.
+	workerPool *workerpool.WorkerPool
+
+	// ticker contains the ticker that is used to send commitment requests.
+	ticker *eventticker.EventTicker[iotago.SlotIndex, iotago.CommitmentID]
+
+	// Logger embeds a logger that can be used to log messages emitted by this chain.
+	log.Logger
+}
+
+// newCommitmentsProtocol creates a new commitment protocol instance for the given protocol.
+func newCommitmentsProtocol(protocol *Protocol) *CommitmentsProtocol {
+	c := &CommitmentsProtocol{
+		Logger:     lo.Return1(protocol.Logger.NewChildLogger("Commitments")),
+		protocol:   protocol,
+		workerPool: protocol.Workers.CreatePool("Commitments"),
+		ticker:     eventticker.New[iotago.SlotIndex, iotago.CommitmentID](protocol.Options.CommitmentRequesterOptions...),
+	}
+
+	c.ticker.Events.Tick.Hook(c.SendRequest)
+
+	return c
+}
+
+// StartTicker starts the ticker for the given commitment.
+func (c *CommitmentsProtocol) StartTicker(commitmentPromise *promise.Promise[*Commitment], commitmentID iotago.CommitmentID) {
+	c.ticker.StartTicker(commitmentID)
+
+	commitmentPromise.OnComplete(func() {
+		c.ticker.StopTicker(commitmentID)
+	})
+}
+
+// SendRequest sends a commitment request for the given commitment ID to all peers.
+func (c *CommitmentsProtocol) SendRequest(commitmentID iotago.CommitmentID) {
+	c.workerPool.Submit(func() {
+		c.protocol.Network.RequestSlotCommitment(commitmentID)
+
+		c.LogDebug("request", "commitment", commitmentID)
+	})
+}
+
+// SendResponse sends a commitment response for the given commitment to the given peer.
+func (c *CommitmentsProtocol) SendResponse(commitment *Commitment, to peer.ID) {
+	c.workerPool.Submit(func() {
+		c.protocol.Network.SendSlotCommitment(commitment.Commitment, to)
+
+		c.LogTrace("sent commitment", "commitment", commitment.LogName(), "toPeer", to)
+	})
+}
+
+// ProcessResponse processes the given commitment response.
+func (c *CommitmentsProtocol) ProcessResponse(commitmentModel *model.Commitment, from peer.ID) {
+	c.workerPool.Submit(func() {
+		if commitment, published, err := c.protocol.Commitments.publishCommitmentModel(commitmentModel); err != nil {
+			c.LogError("failed to process commitment", "fromPeer", from, "err", err)
+		} else if published {
+			c.LogTrace("received response", "commitment", commitment.LogName(), "fromPeer", from)
+		}
+	})
+}
+
+// ProcessRequest processes the given commitment request.
+func (c *CommitmentsProtocol) ProcessRequest(commitmentID iotago.CommitmentID, from peer.ID) {
+	c.workerPool.Submit(func() {
+		commitment, err := c.protocol.Commitments.Get(commitmentID)
```

Comment:
That plays into the realm of how to handle our eviction strategy and yes, I think every node should solidify all commitments up till the genesis.

The easiest solution would be to add a check if the requested slot is below our root commitment and then resolve these requests against the permanent storage of our main engine instead of looking up the matching engine dynamically but this would have to be adressed in an additional PR.